test/built-ins/Atomics/wait/cannot-suspend-throws.js

```diff
@@ -18,9 +18,10 @@ features: [Atomics, SharedArrayBuffer, TypedArray]
 flags: [CanBlockIsFalse]
 ---*/
 
-var buffer = new SharedArrayBuffer(4);
-var int32Array = new Int32Array(buffer);
+const i32a = new Int32Array(
+  new SharedArrayBuffer(Int32Array.BYTES_PER_ELEMENT)
+);
 
 assert.throws(TypeError, function() {
-  Atomics.wait(int32Array, 0, 0, 0);
+  Atomics.wait(i32a, 0, 0, 0);
```

Comment:
this test is not tricky. We set a flag for CanBlock being false, but without any handling, it fails on not throwing.

Reply:
> it fails on not throwing

Correct. It's unclear to me how we can safely communicate to any test consumer that any file, which contain tests for main threads whose host sets `CanBlock === false`, should be ignored if the runtime does not define a main thread that prohibits blocking in the main thread.